### PR TITLE
Misc changes and update to version 1.0.0

### PR DESCRIPTION
**Pull request by khtufteland**
commit 4a5d7e5fee6d2479db9b93d7f4b6bfd57c491876
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Wed Jan 5 14:04:20 2022 +0100

    1.0.0

commit 876b7f886df36b684ac592054ad8291061cf757a
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Wed Jan 5 14:03:47 2022 +0100

    npm: Update dependencies

commit 034a4076022de7c13813b0a6685b8a316e618602
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Fri Jan 7 15:39:10 2022 +0100

    clarify_insert: Don't allow unknown object elements in Signal

commit 9b10e82d60fb99a3534b7e542895382b0ceca80a
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Fri Jan 7 15:44:00 2022 +0100

    clarify_insert: Merge 2 outputs into 1
    
    Did also update the documentation for the node.

commit afafe7ff6ae677adc2835708ebcde35084c2331e
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Wed Jan 5 14:49:50 2022 +0100

    clarify_api: Return only data from Clarify, not Axios meta data

commit 83b82c4e26e50e08622a8e512246f2fa391cad7d
Author: Kjetil Hope Tufteland <kjetil@searis.no>
Date:   Wed Dec 8 15:28:52 2021 +0100

    clarify_api: Change incorrect variable name
    
    Should not use Legacy in variable name.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,13 @@ As for the insert node _output_ message format, development is still ongoing, an
 ## Changelog
 
 The changelog is introduced from `v.1.0.0-beta.4`, and describe changes from `v.1.0.0-beta.3`.
+### 1.0.0
 
+- Change outputs to only output data from Clarify servers.
+- Change Clarify URLs from clarify.us to clarify.io.
+### 1.0.0-beta.9
```
Comment (smyrman) [inline README.md:41]: missing newline
```suggestion

### 1.0.0-beta.9
```

File: README.md [hunk 2]
```diff
@@ -34,7 +34,13 @@ As for the insert node _output_ message format, development is still ongoing, an
...
+
+- Moved code from repo github.com/searis to github.com/clarify.
```
Comment (smyrman) [inline README.md:43]: same

Comment (smyrman) [reply README.md:43]: You can try Prettify plugin to VS-Code to lint the file.

**Review by smyrman — COMMENTED** [review]
Should this PR also reduce to 1 output?

**Comment by smyrman** [comment]
https://github.com/clarify/node-red-contrib-clarify/blob/75de62be576951d945be710d6d07298d63265831/nodes/clarify_insert.html#L18

Should we change this to one?

As for the output fields, the standard http request node returns:
`msg.statusCode` 
`msg.payload` <- JSON if possible, string otherwise

Does it make sense for us to return the same _if_ we have an output? Maybe include _which_ type of request was made?

```
{
  statusCode: 200,
  isError: true
  method: "integration.saveSignals"
  payload: ...
}
```
If you want to treat data differently based on method, isError and/or status, code, you can.


The interesting part to see though, is probably the _message IDs_ of the messages that failed and didn't fail. We can add that later and be backwards compatible.

**Comment by khtufteland** [comment]
@bbergshaven @smyrman 

This PR wasn't meant to remove one of the outputs. For me, the main reason for having two outputs is that I can debug from the first output, and I can connect the second output directly to the Sentry node. If we remove the second output and merge these two outputs into one, must we use an additional block to filter out errors before sending them to Sentry.

I can merge these two outputs, but I don't really see any benefits and it will only create extra work.

**Comment by smyrman** [comment]
> For me, the main reason for having two outputs is that I can debug from the first output, and I can connect the second output directly to the Sentry node. 

Yes, but nobody else does that, right?

**Comment by khtufteland** [comment]
Trying to come up with a new proposal:
When everything is OK we send the following:
```js
{
  "status": 200,
  "statusText": "OK",
  "method": "integration.Insert",
  "headers": { ... },
  "payload": { okDataFromClarify },
  "_msgid": "..."
}
```
when there's an error we send a `node.error(errorAsString, responseAsObject)` which will be fetched by the `CatchAll` block and outputted  with the following format:
```js
{
  "status": 200,
  "statusText": "OK",
  "method": "integration.Insert",
  "headers": { ... },
  "payload": { errorDataFromClarify },
  "error": {
    "message": "...",
    "source": {
      "id": "ce07fa30a23e3452",
      "type": "clarify_insert",
      "name": "insert @ prod/kjetil",
      "count": 1
    }
  },
  "_msgid": "..."
}
```

This message can be transmitted directly to Sentry:
https://sentry.io/organizations/searis/issues/2753925809/?project=5544030&query=is%3Aunresolved

Please note that we now have obtained the trace ID from Clarify.

The insert block we output:
```js
{
  "status": 200,
  "statusText": "OK",
  "method": "integration.Insert",
  "headers": { ... },
  "payload": { errorDataFromClarify },
  "_msgid": "..."
}
```

The problem now is that it's a bit hard to distinguish between errors based on node output. Could introduce `isError = true`, but if we want to be aligned with "Node RED standard" we don't want this flag. Any thoughts, @smyrman ?

**Comment by smyrman** [comment]
The problem now is that it's a bit hard to distinguish between errors based on node output. Could introduce isError = true, but if we want to be aligned with "Node RED standard" we don't want this flag. Any thoughts, @smyrman ?

I like the new output, and yes, I agree, that is the remaining problem. I also agree that isError isn't nice neither.

Maybe we just document output fields properly?

statusCode (number): 200 if there where no transport-layer errors, the HTTP error code otherwise.
payload (object): when status code is 200, contain the JSON RPC API response body. If the operation was sucesssfull, this object will contain an non-empty result field. If the operation failed, the object will contain a non-empty error field. See the API documentation for details [link].

**Comment by smyrman** [comment]
Alternatively, we could do:

```js
{
   errorType: "HTTP",
   statusCode: 401,
   method: "integration.Insert",
   headers: {...}
   payload: {...} // parse JSON body or use string fallback.
}
```
```js
{
   errorType: "RPC",
   statusCode: 200,
   method: "integration.Insert",
   headers: {...},
   payload: {...} // parse JSON body.
}
```

**Comment by smyrman** [comment]
Yet another alternative; only return successful messages (as we have already thrown an error).

**Comment by bbergshaven** [comment]
This looks good. We need to write something in our release note / README.md about changing from 2 to 1 output and the change in format. This is a potentially breaking change if people have logic checking the output.
Should do it anyways, just document it.

**Comment by khtufteland** [comment]
Update the PR, including the Changelog.

The new output format is the following:
- method
- statusCode
- payload
- headers
- errorType

Please note that we now use the method `node.error(errorText, remainingObject)`. This enables us to send the complete error data in `remainObject`. This function will behave differently based on the existence of a CatchAll block.
If there's a CatchAll block available can the user retrieve an object containing the `errorText` and the `remainingObject`, merged into one object. If there isn't any CatchAll block will the `errorText` be presented in the Debug lob, without the `remainingObject`. Decided therefore to always output errors to `msg.payload`.

**Comment by smyrman** [comment]
Acceptable!

**Comment by khtufteland** [comment]
@bbergshaven Updated the PR with the commit `clarify_insert: Don't allow unknown object elements in Signal` that address your finding.

**Review by smyrman — APPROVED** [review]

**Review by smyrman — APPROVED** [review]
Requirements file is out of date (lock file is missnig).

Had to manually install a dependency to get things to work.

**Pull request merged**